### ISSUE TITLE
VR roadmap

### ISSUE THREAD

**Issue by AndersMalmgren** [RobertBeckebans/RBDOOM-3-BFG#925]
Hi, im just checking in on how the state is for VR with PBR textures are coming along.
Cheers

**Comment by RobertBeckebans** [comment]
Well the renderer is working with OpenVR in the mods/samsonvr branch. I actually put a lot of work into it by cleaning up and refactoring the original code. Right now it is a stripped down version that does not use any of the Fully Possessed assets and it lacks motion controller support. It's like OpenVR but for the raw BFG edition and plays well with either gamepad or keyboard + mouse. Renderer wise only TAA is lacking. So the focus so far was to get the iVr class going with DX12/Vulkan instead of OpenGL. Since RBDoom 1.5 got rid of stencil shadows I also made some optimizations and changed the entire renderloop in the backend so shadow maps are only rendered into an atlas for the first eye and can be reused for the lighting by the second eye. 

So renderer wise it is concerning OpenVR support almost finished and if we want to have the rest of the features of the original code then it is mainly just merging the game code. However that would cost me approx 2-3 fulltime weeks to encounter unknown issues. I don't know when I can find that time.

**Comment by AndersMalmgren** [comment]
> Hey since RBDoom 1.6 is finally finished and there is only some minor Linux stuff going on VR will be the next major thing.
> 
> I started a new project board so you can watch the process: [RBDOOM-VR 1.0](https://github.com/users/RobertBeckebans/projects/3)
> 
> It won't be as massive as the Fully Possessed mod was but I want to make it a fully playable and smooth experience with less options. However the controls should be more standard like in HL:Alyx and Half-Life 2 VR.
> 
> This branch will be based on Leyland's OpenVR branch instead of Samson's because it is less bloated and much easier to extend. It is already playable with some minor bugs.

Thats excellent news, some important options to me from Fully possed that I hope can get it into your port. 

Crouch to height, meaning my IRL floor matches up to the floor in doom. 
World scale: Doom 3 scale is off. I like 0.89 its a good balance between too small hands and a too large world. 
No body, the IK is buggy and laggy and doesnt add anything. 
Dot lasersight instead of line, even cooler would be working reflex sights etc, but I understand that could be a stretch. 
Disable hud, that floating hud in fully possed takes me out of the imerssion, the watch is enough.

**Comment by RobertBeckebans** [comment]
Yeah the first 3 issues were also really annoying for me and that is already solved. 
You can set vr_playerHeightCM and that will do the scaling for you.

**Comment by AndersMalmgren** [comment]
> Yeah the first 3 issues were also really annoying for me and that is already solved. You can set vr_playerHeightCM and that will do the scaling for you.

Is really world scale a function of player height? I mean the size of objects are the same regardless of person :P
